astropy/io/fits/hdu/hdulist.py

```diff
@@ -1517,6 +1518,8 @@ def _flush_resize(self):
                 # lead to odd behavior in practice.  Better to just not keep
                 # references to data from files that had to be resized upon
                 # flushing (on Windows--again, this is no problem on Linux).
+                # Note that this hack is only possible on numpy 1.x:
+                # in 2.x, we cannot write directly to the data attribute
                 for idx, mmap, arr in mmaps:
```

Comment:
We should add a warning if there are references to the mmap array, accessing it will segault ! (which was the reason for this hack).

Reply:
Good point. I'll give this a try now.

Reply:
here it is https://github.com/astropy/astropy/pull/16608
please let me know if I've accurately implemented what you had in mind